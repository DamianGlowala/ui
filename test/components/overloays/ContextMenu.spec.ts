import { describe, it, expect } from 'vitest'
import { UContextMenu } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('ContextMenu', () => {
  it.each([
    ['basic case', {}],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof ContextMenu.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, UContextMenu)

    expect(html).toMatchSnapshot()
  })
})
