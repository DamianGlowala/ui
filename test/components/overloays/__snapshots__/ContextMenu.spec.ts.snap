// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContextMenu > renders basic case correctly 1`] = `
"<div class="z-20 group">
  <transition-stub appear="true" persisted="false" css="true">
    <div>
      <!--v-if-->
      <div class="">label</div>
    </div>
  </transition-stub>
</div>"
`;

exports[`ContextMenu > renders with popper correctly 1`] = `
"<div class="z-20 group">
  <transition-stub appear="true" persisted="false" css="true">
    <div>
      <!--v-if-->
      <div class="">label</div>
    </div>
  </transition-stub>
</div>"
`;
